### PR TITLE
Add MobSF producer

### PR DESCRIPTION
**Pull request by chrisnovakovic**
Support [MobSF](https://github.com/MobSF/Mobile-Security-Framework-MobSF) as a Dracon producer. Since MobSF runs as a server and exposes a REST API for uploading mobile app projects to be scanned, this requires the MobSF container to be vendorised and integrated into Dracon with a wrapper that initialises MobSF, identifies individual MobSF-compatible mobile app projects within the target code base, compresses and uploads these projects to MobSF, and retrieves and filters scan reports, as well as doing the usual producer tasks.

~(Work in progress: the tool container and wrapper are finished, but the producer isn't.)~ Since the distinction between tool and producer in this case is blurred, and because drawing an arbitrary line between them makes the data interchange format unnecessarily difficult to define, I'm calling the whole thing a producer to simplify things.

### CODE REVIEW

File: producers/mobsf/report/ios/ios.go [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+	seen := make(map[string]bool)
+	for _, finding := range r.ATSAnalysis {
+		if _, exists := seen[finding.Issue]; !exists {
+			seen[finding.Issue] = true
+
+			issue := &v1.Issue{
+				// MobSF doesn't report the precise source of the issue so we
+				// can't be more specific about its location than this:
+				Target:      r.RootDir,
+				Type:        "Insecure App Transport Security policy",
```
Comment (northdpole) [inline producers/mobsf/report/ios/ios.go:82]: is this type always the case? or maybe a copy paste error?

File: producers/mobsf/report/ios/ios.go [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+				Title:       finding.Issue,
+				Confidence:  v1.Confidence_CONFIDENCE_MEDIUM,
+				Description: fmt.Sprintf(
+					"An insecure App Transport Security policy is defined in a plist file in the iOS app project directory %s.\n\nDetails:\n\n%s\n%s",
```
Comment (northdpole) [inline producers/mobsf/report/ios/ios.go:86]: same here

**Review by northdpole — CHANGES_REQUESTED** [review]

File: producers/mobsf/report/ios/ios.go (hunk 1 above)
Comment (chrisnovakovic) [reply producers/mobsf/report/ios/ios.go:82]: Yes, the ATS analysis always produces reports of this type - the specific violation is given in `Title`. The report only contains three fields: `Issue` (a one-line summary), `Description` (sometimes one word, sometimes a whole paragraph), and `Status` (which I'm using as a proxy for severity).

File: producers/mobsf/report/ios/ios.go (hunk 2 above)
Comment (chrisnovakovic) [reply producers/mobsf/report/ios/ios.go:86]: Same as above

**Review by northdpole — APPROVED** [review]

**Pull request merged**